test/test_sparse.py

```diff
@@ -3248,33 +3248,59 @@ def test_cuda_sparse_cpu_dense_add(self):
         with self.assertRaisesRegex(RuntimeError, "add: expected 'self' to be a CUDA tensor, but got a CPU tensor"):
             x + sparse_y
 
-class TestSparseUnaryUfuncs(TestCase):
+class TestSparseFuncs(TestCase):
     exact_dtype = True
 
+    # TODO - Need to extend for non-unary functions
     @ops(sparse_unary_ufuncs)
     def test_sparse_consistency(self, device, dtype, op):
-        unsupportedTypes = [torch.bfloat16, torch.cfloat, torch.cdouble]
+        unsupportedTypes = [torch.bfloat16, torch.cfloat, torch.cdouble, torch.bool]
         if dtype in unsupportedTypes:
             self.skipTest('Skipped! Unsupported dtypes for Sparse')
 
-        samples = op.sample_inputs(device, dtype)
+        samples = op.sample_sparse_inputs(device, dtype)
 
         if len(samples) == 0:
             self.skipTest("Skipped! No sample inputs!")
 
-        sample = samples[0]
+        for sample in samples:
+            sample = sample.input[0]
+            expected = op(sample.to_dense())
+            output = op(sample)
+            self.assertEqual(output.to_dense(), expected)
 
-        if len(sample.input) > 1:
-            self.skipTest("Skipped! Testing unary ops, one input is expected")
-        sample = sample.input[0]
+    # TODO - Need to extend for non-unary functions
+    @ops(sparse_unary_ufuncs)
+    def test_sparse_inplace_consistency(self, device, dtype, op):
+        unsupportedTypes = [torch.bfloat16, torch.cfloat, torch.cdouble, torch.bool]
+        if dtype in unsupportedTypes:
+            self.skipTest('Skipped! Unsupported dtypes for Sparse')
 
-        expected = op(sample)
-        assert torch.is_tensor(expected)
-        output = op(sample.to_sparse())
-        assert torch.is_tensor(output)
-        self.assertEqual(output.to_dense(), expected)
+        inplace_variant = op.get_inplace()
+
+        if inplace_variant is not None:
+            samples = op.sample_sparse_inputs(device, dtype)
+            if len(samples) == 0:
+                self.skipTest("Skipped! No sample inputs!")
+
+            for sample in samples:
+                # TODO - extend this logic for Binary Operations
+                sample = sample.input[0]
+                sample_copy = sample.clone()
+                expected = op(sample.to_dense())
+
+                if torch.can_cast(expected.dtype, sample.dtype) and (
+                        sample.is_coalesced() or op.is_linear_map):
+                    inplace_variant(sample)
+                    self.assertEqual(expected, sample.to_dense())
+                    op(sample_copy, out=sample_copy)
+                    self.assertEqual(sample, sample_copy)
+                else:
+                    with self.assertRaises(RuntimeError):
```

Comment:
Nit: I'd say it is better to catch a single exception to test for. Here it should catch 2 types of exceptions: one for promoting dtypes and second for inplace on uncoalesced with the op unsupporting inplace on uncoalesced (not is_linear_map here).

Reply:
> Nit: I'd say it is better to catch a single exception to test for. Here it should catch 2 types of exceptions: one for promoting dtypes and second for inplace on uncoalesced with the op unsupporting inplace on uncoalesced (not is_linear_map here).

Reply:
@vfdev-5  - I understand the concern raised here; however, both cases are handled with the same runtime error; as this framework can be used for testing many ops; this way we are limiting the Op implementor only to raise a Runtime error leaving the specifics of the message to the developer.
@mruberry  let me know your opinion about this

Reply:
Both arguments are reasonable, but I don't think we need to block this PR on a resolution since it's principally about supporting asinh. We can continue to revise and improve this test in the future.

Reply:
@mruberry - Thanks for the confirmation; let me know if any other changes are required in this PR.